### PR TITLE
feat(DNA-1716): Flash Recovery

### PR DESCRIPTION
**Pull request by SamraHanifCareem**
- [x] Conditional Actions when flash status is DELETED
- [x] Flash Recovery action on Grid
- [x] Confirmation Modal along with necessary validations for Flash Recovery
- [x] Api integration
- [x] Refreshing the Grid Data after update
- [x] Adding ttl in payload on the basis of flash type

### CODE REVIEW

File: superset-frontend/src/views/CRUD/flash/components/FlashGrid/FlashList.tsx [hunk 1]
```diff
@@ -150,6 +162,44 @@ function FlashList({ addDangerToast, addSuccessToast }: FlashListProps) {
...
+    let currentTtl = flash?.ttl;
+    const maxDate = new Date(flash.ttl);
+
+    const flashType = flash.flashType.replace(/([A-Z])/g, ' $1').trim();
+    if (
+      flashType === FlashTypes.SHORT_TERM ||
+      flashType === FlashTypes.ONE_TIME
+    ) {
+      maxDate.setDate(maxDate.getDate() + 7);
+      currentTtl = new Date(maxDate).toISOString().split('T')[0];
```
Comment (puneet-bdp) [inline superset-frontend/src/views/CRUD/flash/components/FlashGrid/FlashList.tsx:175]: Why are we sending the date string without timezone information?

File: superset-frontend/src/views/CRUD/flash/components/FlashGrid/FlashList.tsx [hunk 2]
```diff
@@ -150,6 +162,44 @@ function FlashList({ addDangerToast, addSuccessToast }: FlashListProps) {
...
+          refreshData();
+        },
+        createErrorHandler(errMsg =>
+          addDangerToast(
+            t('There was an issue recovering %s: %s', flash?.tableName, errMsg),
+          ),
+        ),
+      );
+    } else {
+      addDangerToast('There is an issue recovering the selected flash');
```
Comment (puneet-bdp) [inline superset-frontend/src/views/CRUD/flash/components/FlashGrid/FlashList.tsx:197]: Could we add some informative error message? Like flash id missing or something?

File: superset-frontend/src/views/CRUD/flash/components/FlashGrid/FlashList.tsx [hunk 3]
```diff
@@ -150,6 +162,44 @@ function FlashList({ addDangerToast, addSuccessToast }: FlashListProps) {
...
+    if (flash && flash?.id) {
+      recoverFlashObject(flash?.id, payload).then(
+        () => {
+          setRecoverFlash(null);
+          addSuccessToast(t('Recovered: %s', flash?.tableName));
+          refreshData();
+        },
+        createErrorHandler(errMsg =>
+          addDangerToast(
+            t('There was an issue recovering %s: %s', flash?.tableName, errMsg),
```
Comment (puneet-bdp) [inline superset-frontend/src/views/CRUD/flash/components/FlashGrid/FlashList.tsx:192]: Will this error message contain the exception information thrown from backend?

File: superset-frontend/src/views/CRUD/flash/components/FlashGrid/FlashList.tsx (hunk 1 above)
Comment (SamraHanifCareem) [reply superset-frontend/src/views/CRUD/flash/components/FlashGrid/FlashList.tsx:175]: yes, we are sending without timezone.

File: superset-frontend/src/views/CRUD/flash/components/FlashGrid/FlashList.tsx (hunk 3 above)
Comment (SamraHanifCareem) [reply superset-frontend/src/views/CRUD/flash/components/FlashGrid/FlashList.tsx:192]: Yes, along with the flash name

File: superset-frontend/src/views/CRUD/flash/components/FlashGrid/FlashList.tsx (hunk 2 above)
Comment (SamraHanifCareem) [reply superset-frontend/src/views/CRUD/flash/components/FlashGrid/FlashList.tsx]: Missing parameter added in error messsage

**Review by puneet-bdp — APPROVED** [review]

**Review by puneet-bdp — APPROVED** [review]

**Pull request merged**